### PR TITLE
sjasmplus: init at 1.18.2

### PR DESCRIPTION
**Pull request by electrified**
###### Motivation for this change
sjasmplus is a popular Z80 assembler, used by people coding for "retro" platforms (e.g. Gameboy, Sega Master System, ZX Spectrum etc). It is maintained and under current development, with relatively frequent releases. I think it would be a good addition to nixpkgs.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 116671` at 2f0071e6 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/c66f6e5097f35a778de95e5119763f10">sjasmplus</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/compilers/sjasmplus/default.nix:14:3:
```
   |
14 |   installPhase = "install -D sjasmplus $out/bin/sjasmplus";
   |   ^
```
</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -2725,6 +2725,12 @@
     githubId = 1753498;
     name = "Dejan Lukan";
   };
+  electrified = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:2728]: Please put this into a separate commit with the message ``maintainers: add electrified``.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 116671` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>sjasmplus: <a href="https://termbin.com/bgae5">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/82be105e393298569637931c9ca83ba6/raw/index.html">pretty log </a></li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 116671` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>sjasmplus</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
sjasmplus:
<details>

Using CC=cc, CXX=c++, AR=ar or LD=cc prevents cross compiling.
Please use ``CC=${stdenv.cc.targetPrefix}cc``, ``CXX=${stdenv.cc.targetPrefix}c++``, ``AR=${stdenv.cc.targetPrefix}ar`` or ``LD=${stdenv.cc.targetPrefix}cc``.

</details>
</details>
